### PR TITLE
Font dialog

### LINKED ISSUE #1080: Font Dialog for new Font classes
There should be a dialog class that allows a user to select a font and return a Pyface `Font` object.

### PR DESCRIPTION
**Pull request by corranwebster**
This adds a standard font selection dialog class that works with `Font` classes.

This requires #1042 to be merged first.

Fixes #1080

### CODE REVIEW

**Review by flongford — COMMENTED** [review]
@corranwebster - this may be a bit of a shallow review but I've pulled the branch, ran the unit tests locally on MacOS (qt5) and also tested the example code provide. 

Everything seems to be working as expected, so I just have a couple of minor questions.

Let me know if there is an other area that would be beneficial to focus on

File: pyface/tests/test_font_dialog.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    @unittest.skipIf(no_modal_dialog_tester, "ModalDialogTester unavailable")
+    def test_close_show_alpha(self):
+        # test that cancel works as expected
+        tester = ModalDialogTester(
+            lambda: get_font(None, "12 Helvetica sans-serif")
+        )
+        tester.open_and_run(when_opened=lambda x: x.close(accept=False))
+
+        self.assertEqual(tester.result, None)
```
[the remark this answers is not in the document]
Comment (corranwebster) [reply pyface/tests/test_font_dialog.py:97]: That's a copy-pasta from the color dialog tests.  So this needs fixing.

File: pyface/font_dialog.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+"""
+
+from .constant import OK
+from .toolkit import toolkit_object
+
+
+FontDialog = toolkit_object("font_dialog:FontDialog")
+
+
+def get_font(parent, font):
```
[the remark this answers is not in the document]
Comment (corranwebster) [reply pyface/font_dialog.py:21]: I think so - this is a convenience and so if it doesn't fit then a developer can always use the class instead.

For the `parent` argument, I'd like the user to at least think a minute, as we've had some issues reported where the parent was None when it should have been something else, so I don't want a default for that.

And if in practice this is a problem, we can always add defaults later - but better to have no default than to have the wrong one.

**Comment by corranwebster** [comment]
Getting failures from MacOS/PySide6. Need to look into it.

**Comment by corranwebster** [comment]
> Let me know if there is an other area that would be beneficial to focus on

This is a fairly simple PR, so there isn't much else to it (the heavy lifting is already in place).

I think everything is addressed and we have tests back to working state, so this just needs merge approval.

**Review by flongford — APPROVED** [review]
LGTM!

**Pull request merged**